config/v1/types_clustercli.go

```diff
@@ -0,0 +1,56 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ClusterCLI is the Schema for the cluster cli API
+type ClusterCLI struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// +kubebuilder:validation:Required
+	// +required
+	Spec ClusterCLISpec `json:"spec"`
+	// status holds observed values from the cluster. They may not be overridden.
+	// +optional
+	Status ClusterCLIStatus `json:"status,omitempty"`
+}
+
+// ClusterCLISpec defines the desired state of ClusterCLI
+type ClusterCLISpec struct {
+	// Description of ClusterCLI
+	Description string `json:"description"`
+	// DisplayName for CLI
+	DisplayName string `json:"displayName"`
+	// Image is the cli image that contains the cli artifacts
+	Image string `json:"image"`
+	// Mapping defines extract targets for ClusterCLIs
+	Mapping []ClusterCLIMapping `json:"mapping,omitempty"`
+}
```

Comment:
Need to add a field for where the artifacts are published, for example, most of our current CLIs will populate with 
https://mirror.openshift.com/pub/openshift-v4/clients/